### ISSUE TITLE
drone.pair() bug?

### ISSUE THREAD

**Issue by bryanperez43** [RobolinkInc/CoDrone-python#14]
not entirely sure if i have a bad BLE board or if there is an issue with the pair function

```
import CoDrone

drone = CoDrone.CoDrone()

# drone.pair(drone.Nearest)   # should be done only once
drone.pair()

drone.takeoff()

...
```

The drone pairs perfectly using the `drone.Nearest` method but fails to find the drone with just `drone.pair()`

**Comment by RobolinkAdmin** [comment]
Hello,

Thank you for reaching out to us.

We are sorry to hear that you're having issues with pairing. We would like
to know first if this is for CoDrone Pro, Lite or Mini before we transfer
you to our technical team.

Thanks,
Cha



*Robolink Inc. *





*www.robolink.com
<http://www.robolink.com/>Facebook: /robolinkincTwitter:
@robolinkincInstagram: @robolinkinc*
*Phone: 1-858-876-5123*
*Address: 5677 Oberlin Dr. Ste 100, San Diego, CA 92121*

**Comment by bryanperez43** [comment]
apologies, it is the CoDrone Lite

**Comment by bryanperez43** [comment]
apologies, CoDrone Lite

**Comment by RobolinkAdmin** [comment]
Hello,

Thank you for the confirmation.

I have cc'd our technical engineer, Leila, on this email for you to be
assisted further on your issue with CoDrone Lite pairing.
She will be in contact with you within the next few days.

Best,
Cha

**Comment by firestoneleila** [comment]
@bryanperez43 Try using drone.pair() with the drone ID number. For example, drone.pair("1234"). You can find the ID of the drone when pairing with drone.Nearest and looking at the console. Let us know if you are still having trouble. We will look into the issue further!

**Comment by ArN5** [comment]
@bryanperez43 seems like the code may be missing drone.close(). 
Also, make sure the LED color on the BLE board is blinking red and not pulsing yellow or green when trying to pair with python.

If it is green then the pairing method won't work since the serial port is busy from a prior connection to CoDrone. If you are in this green led state then unplug and replug the BLE board over microusb, then click the small button on the BLE board until you get the red led blinking.

**Comment by bryanperez43** [comment]
> @bryanperez43 Try using drone.pair() with the drone ID number. For example, drone.pair("1234"). You can find the ID of the drone when pairing with drone.Nearest and looking at the console. Let us know if you are still having trouble. We will look into the issue further!


I have noticed that using drone.connect(device_name='id')
Works a lot better than using pair(). Using pair takes wayy too long

**Comment by bryanperez43** [comment]
> @bryanperez43 seems like the code may be missing drone.close(). 
> 
> Also, make sure the LED color on the BLE board is blinking red and not pulsing yellow or green when trying to pair with python.
> 
> 
> 
> If it is green then the pairing method won't work since the serial port is busy from a prior connection to CoDrone. If you are in this green led state then unplug and replug the BLE board over microusb, then click the small button on the BLE board until you get the red led blinking.

Yes I've noticed that while reading the different functions. I am just surprised that the CoDrone website instruction don't mention it

**Comment by ArN5** [comment]
@bryanperez43 can you try out this code and let me know what the average time for your CoDrone is. 
Here are my results "Avg time with connect =  16.05744848251343  pair =  7.568765878677368"
```
#Python code
import CoDrone
import CoDrone.protocol
import time


drone_number = "4640"
iterations = 5

attempts = 0
time_avg = 0
avg1 = 0
avg2 = 0

for i in range(iterations):
    t1= time.time()
    drone = CoDrone.CoDrone()
    try:
        drone.connect(drone_number)
        attempts = attempts + 1
        time_avg = time_avg + (time.time()-t1)
        print("---------",attempts, " ---" , time.time()-t1)
    finally:
        drone.close()
time_avg = time_avg/iterations
avg1 = time_avg



attempts = 0
time_avg = 0
for i in range(iterations):
    t1= time.time()
    drone = CoDrone.CoDrone()
    try:
        drone.pair(drone_number)
        attempts = attempts + 1
        time_avg = time_avg + (time.time()-t1)

        print("---------",attempts, " ---" , time.time()-t1)
    finally:
        drone.close()

time_avg =  time_avg/iterations
avg2 = time_avg

print("Avg time with connect = " , avg1, " pair = ", avg2)
```
```

**Comment by bryanperez43** [comment]
> @bryanperez43 can you try out this code and let me know what the average time for your CoDrone is.
> Here are my results "Avg time with connect = 16.05744848251343 pair = 7.568765878677368"
> 
> ```
> #Python code
> import CoDrone
> import CoDrone.protocol
> import time
> 
> 
> drone_number = "4640"
> iterations = 5
> 
> attempts = 0
> time_avg = 0
> avg1 = 0
> avg2 = 0
> 
> for i in range(iterations):
>     t1= time.time()
>     drone = CoDrone.CoDrone()
>     try:
>         drone.connect(drone_number)
>         attempts = attempts + 1
>         time_avg = time_avg + (time.time()-t1)
>         print("---------",attempts, " ---" , time.time()-t1)
>     finally:
>         drone.close()
> time_avg = time_avg/iterations
> avg1 = time_avg
> 
> 
> 
> attempts = 0
> time_avg = 0
> for i in range(iterations):
>     t1= time.time()
>     drone = CoDrone.CoDrone()
>     try:
>         drone.pair(drone_number)
>         attempts = attempts + 1
>         time_avg = time_avg + (time.time()-t1)
> 
>         print("---------",attempts, " ---" , time.time()-t1)
>     finally:
>         drone.close()
> 
> time_avg =  time_avg/iterations
> avg2 = time_avg
> 
> print("Avg time with connect = " , avg1, " pair = ", avg2)
> ```
 
using pair('devive_#') may be faster 
however try using it a few times for just taking off and hover and you will notice how it reports a 0% battery
when clearly its not at 0% since the led lights are still going. 
then try using connect('device_#') and you will see that the battery is actually not 0%
